File: rust/eu002.rs
fn fibsum(max: i32) -> i32 {
    let (mut f0, mut f1, mut r) = (1,2,2);
    while f1 < max {
        let f2 = f0 + f1;
        if f2 < max && f2%2 == 0 {
            r += f2;
        }
        f0 = f1;
        f1 = f2;
    }
    println!("sum of even fibonacci numbers under {}: {}",max,r);
    r
}

fn main() {
    if fibsum(90) != 44 {
        panic!("test failed! expecting result: 44}");
    }
    fibsum(4000000);
}

File: zig/eu002.zig
const std = @import("std");
const expect = @import("std").testing.expect;

fn fibsum(max: i32) i32 {
    var fib0: i32 = 1;
    var fib1: i32 = 2;
    var fibNext: i32 = 3;
    var result: i32 = 2;
    while (fib1 < max) {
        fibNext = fib0 + fib1;
        if (fibNext < max and @mod(fibNext,2) == 0) {
            result += fibNext;
        }
        fib0 = fib1;
        fib1 = fibNext;
    }
    return result;
}

test "sum of even terms of fibonacci seq below 90" {
    expect(fibsum(90)==44);
}

pub fn main() void {
    std.debug.print("{}\n",.{fibsum(@as(i32,4e6))});
}
